core/domain/summary_services.py

```diff
@@ -124,10 +125,21 @@ def _generate_query(categories):
         if not summary_dicts:
             continue
 
+        for exploration_summary in summary_dicts:
+            if exploration_summary['status'] == 'publicized':
+                featured.append(exploration_summary)
+
         results.append({
             'header': gallery_group['header'],
             'categories': gallery_group['search_categories'],
             'activity_summary_dicts': summary_dicts,
         })
 
+    if featured:
```

Comment:
The featured explorations returned here are those found in the category groupings above, but it's actually possible for featured explorations that are not in any of these categories to exist. For example, we query only 10 explorations per category in line 120. If there's a featured exploration and it's not one of the first 10, it won't be included here.

So, instead of pulling the results from the individual category groupings, I think the function to find the featured explorations needs to be entirely separate, and ideally it should be a different function altogether in this file. It returns its own dict, and is completely independent of this function.